### PR TITLE
Update GitHub Actions' workflows

### PR DESCRIPTION
**Pull request by nnsnodnb**
- Updated steps.
- Improved getting full python version from `actions/setup-python@4`.
- `set-output` was deprecate.
- `poetry install` added `--sync` argument.

Thank you!

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -43,18 +40,18 @@ jobs:
...
-          key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
 
       - name: Ensure cache is healthy
         if: steps.cache.outputs.cache-hit == 'true'
         run: timeout 10s poetry run pip --version || rm -rf .venv
 
       - name: Install dependencies
-        run: poetry install --no-root
+        run: poetry install --sync --no-root
```
Comment (mpeteuil) [inline .github/workflows/build.yml:54]: This project does not use a lock file because it's a library, so `--sync` doesn't work here.

**Review by mpeteuil — CHANGES_REQUESTED** [review]
Hey @nnsnodnb! Thanks for the contributions to update these deprecations! There's one change that needs to be made, otherwise this looks.

File: .github/workflows/build.yml (hunk 1 above)
Comment (nnsnodnb) [reply .github/workflows/build.yml]: Thank you for review.
Delete `--sync` flag for poetry.

**Comment by mpeteuil** [comment]
Thanks @nnsnodnb. The reason this was failing was outlined and resolved in #21. If you either rebase against main or merge main into these changes then these tests should run as expected and we can get your contributions merged 😄

**Comment by nnsnodnb** [comment]
Thank you for resolving.
I have rebased it, so please check again.

**Review by mpeteuil — APPROVED** [review]
Thanks again for updating these!

**Pull request merged**